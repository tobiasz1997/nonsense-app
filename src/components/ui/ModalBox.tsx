import { XMarkIcon } from '@heroicons/react/24/outline';
import { FC, PropsWithChildren } from 'react';

type Props = {
	onClose: () => void;
};

const ModalBox: FC<Props & PropsWithChildren> = (props) => {
	return (
		<div className="m-3 w-full max-w-screen-sm">
			<div className="relative rounded-xl bg-pistachio dark:bg-gray-500 p-5 dark:border dark:border-gray-700">
				<div className="absolute top-3 right-3">
					<button
						onClick={props.onClose}
						className="flex items-center justify-center p-1 rounded focus:outline-none focus:ring-2 focus:ring-orange"
					>
						<XMarkIcon className="h-5 w-5 text-orange" />
					</button>
				</div>
				{props.children}
			</div>
		</div>
	);
};

export default ModalBox;
