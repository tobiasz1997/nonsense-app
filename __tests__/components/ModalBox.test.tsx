import { render, screen } from '@testing-library/react';
import ModalBox from '@components/ui/ModalBox';

describe('Modal Box', () => {
	const content = 'content';
	const action = jest.fn(() => null);

	it('should render without crashing', () => {
		const { container } = render(<ModalBox onClose={action} />);
		expect(container).toMatchSnapshot();
	});

	it('should content be not visible', () => {
		render(<ModalBox onClose={action}>{content}</ModalBox>);

		const contentInDoc = screen.queryByText(content);

		expect(contentInDoc).toBeInTheDocument();
	});
});
