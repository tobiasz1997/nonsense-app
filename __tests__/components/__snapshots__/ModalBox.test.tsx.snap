// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Box should render without crashing 1`] = `
<div>
  <div
    class="m-3 w-full max-w-screen-sm"
  >
    <div
      class="relative rounded-xl bg-pistachio dark:bg-gray-500 p-5 dark:border dark:border-gray-700"
    >
      <div
        class="absolute top-3 right-3"
      >
        <button
          class="flex items-center justify-center p-1 rounded focus:outline-none focus:ring-2 focus:ring-orange"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5 text-orange"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18 18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
